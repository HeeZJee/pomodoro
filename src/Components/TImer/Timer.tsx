import {
  buildStyles,
  CircularProgressbarWithChildren,
} from "react-circular-progressbar";
import { Wrapper } from "./Timer.style";

interface iProps {
  minute: number;
  second: number;
  counter: number;
}

const Timer = ({ minute, second, counter }: iProps) => {
  return (
    <Wrapper data-testid="timer">
      <CircularProgressbarWithChildren
        minValue={0}
        value={counter}
        maxValue={1500}
        styles={buildStyles({
          pathTransition: "all",
          pathTransitionDuration: 1,
          strokeLinecap: "round",
          pathColor: `rgba(106, 89, 157, 1)`,
          trailColor: "rgba(106, 89, 157, 0.4)",
        })}
      >
        <p>
          <span data-testid="minute">{minute > 9 ? minute : `0${minute}`}</span>
          :
          <span data-testid="second">{second > 9 ? second : `0${second}`}</span>
        </p>
      </CircularProgressbarWithChildren>
    </Wrapper>
  );
};

export default Timer;
