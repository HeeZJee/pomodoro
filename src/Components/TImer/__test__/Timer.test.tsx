import React from "react";
import { cleanup, render } from "@testing-library/react";
import Timer from "../Timer";

afterEach(cleanup);

it("snapshot Timer", () => {
  const { getByTestId } = render(<Timer minute={25} second={0} />);
  expect(getByTestId("timer")).toMatchSnapshot();
});

it("render Timer", () => {
  const { getByTestId } = render(<Timer minute={25} second={0} />);
  expect(getByTestId("timer")).toBeInTheDocument();
});

it("minute should be between 0 and 25", () => {
  const { getByTestId } = render(<Timer minute={25} second={0} />);
  expect(parseInt(getByTestId("minute").innerHTML)).toBeLessThanOrEqual(25);
  expect(parseInt(getByTestId("minute").innerHTML)).toBeGreaterThanOrEqual(0);
});

it("second should be between 0 and 25", () => {
  const { getByTestId } = render(<Timer minute={25} second={0} />);
  expect(parseInt(getByTestId("second").innerHTML)).toBeLessThanOrEqual(59);
  expect(parseInt(getByTestId("second").innerHTML)).toBeGreaterThanOrEqual(0);
});
