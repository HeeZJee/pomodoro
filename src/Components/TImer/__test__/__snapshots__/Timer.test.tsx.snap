// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Timer 1`] = `
<div
  class="sc-bdfBwQ fJanuH"
  data-testid="timer"
>
  <div
    data-test-id="CircularProgressbarWithChildren"
  >
    <div
      style="position: relative; width: 100%; height: 100%;"
    >
      <svg
        class="CircularProgressbar "
        data-test-id="CircularProgressbar"
        viewBox="0 0 100 100"
      >
        <path
          class="CircularProgressbar-trail"
          d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
          fill-opacity="0"
          stroke-width="8"
          style="stroke: rgba(106, 89, 157, 0.3); stroke-linecap: round; stroke-dasharray: 289.02652413026095px 289.02652413026095px; stroke-dashoffset: 0px;"
        />
        <path
          class="CircularProgressbar-path"
          d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
          fill-opacity="0"
          stroke-width="8"
          style="stroke: rgba(106, 89, 157, 1); stroke-linecap: round; transition: none; transition-duration: 60s; stroke-dasharray: 289.02652413026095px 289.02652413026095px; stroke-dashoffset: 11.561060965210448px;"
        />
      </svg>
      <div
        data-test-id="CircularProgressbarWithChildren__children"
        style="position: absolute; width: 100%; height: 100%; margin-top: -100%; display: flex; flex-direction: column; justify-content: center; align-items: center;"
      >
        <p>
          <span
            data-testid="minute"
          >
            25
          </span>
          :
          <span
            data-testid="second"
          >
            0
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;
